src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -695,6 +695,33 @@ public void combineAllCommitsOfRepositoryIntoOne(VcsRepositoryUrl repoUrl) throw
         gitService.combineAllCommitsIntoInitialCommit(exerciseRepository);
     }
 
+    /**
+     * Updates the timeline attributes of the given programming exercise
+     * @param updatedProgrammingExercise containing the changes that have to be saved
+     * @param notificationText optional text for a notification to all students about the update
+     * @return the updated ProgrammingExercise object.
+     */
+    public ProgrammingExercise updateTimeline(ProgrammingExercise updatedProgrammingExercise, @Nullable String notificationText) {
+
+        Optional<ProgrammingExercise> programmingExercise = programmingExerciseRepository.findById(updatedProgrammingExercise.getId());
+        if (programmingExercise.isPresent()) {
+            programmingExercise.get().setReleaseDate(updatedProgrammingExercise.getReleaseDate());
```

Comment:
You could technically avoid the multiple `.get()` calls by creating a local variable :D

Reply:
You can do something like:
```java
Optional<ProgrammingExercise> programmingExerciseOrEmpty = programmingExerciseRepository.findById(updatedProgrammingExercise.getId());

if(programmingExerciseOrEmpty.isEmpty()) {
   throw new EntityNotFoundException("Programming exercise not found with id: " + updatedProgrammingExercise.getId());
}

var programmingExercise = programmingExerciseOrEmpty.get();

// The rest goes here

```